### PR TITLE
Feat/io2021 c 17 employee supervisor insert form

### PR DESCRIPTION
**Pull request by Pinioo**
Goals:
- Providing user insert mechanism [front]
- Providing sufficient interface for user insertion [back]

### CODE REVIEW

File: spring-app/src/main/java/com/agh/hr/persistence/service/UserService.java [hunk 1]
```diff
@@ -49,15 +51,20 @@ public UserService(UserRepository userRepository,RoleService roleService, Conver
         return result.map(converters::userToDTO);
     }
 
-    public Optional<UserDTO> saveUser(UserDTO userDTO) {
+    public Optional<UserDTO> saveUser(UserInsertionDTO userDTO) {
         User user=converters.DTOToUser(userDTO);
+        System.out.println(user.toString());
```
Comment (ChrisMisan) [inline spring-app/src/main/java/com/agh/hr/persistence/service/UserService.java]: delete

File: spring-app/src/main/java/com/agh/hr/persistence/service/UserService.java [hunk 2]
```diff
@@ -49,15 +51,20 @@ public UserService(UserRepository userRepository,RoleService roleService, Conver
...
 
         if(!roleService.isAdmin(userAuth))
             if(!Auth.getAdd(userAuth))
                 return Optional.empty();
 
         user.setId(0L);
+        user.setPasswordHash(passwordEncoder.encode(userDTO.getPassword()));
+        user.setEnabled(true);
+        user.setAuthorities(Collections.singleton(roleService.employeeRole()));
+        user.setPermissions(Permission.builder().add(false).read(null).write(null).build());
```
Comment (ChrisMisan) [inline spring-app/src/main/java/com/agh/hr/persistence/service/UserService.java]: Extract this to corresponding converter.

**Review by ChrisMisan — CHANGES_REQUESTED** [review]
Reviewed backend part only.

File: react-app/src/components/Profile/UserForm/UserFormDate.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import { MaterialUiPickersDate } from '@material-ui/pickers/typings/date';
+
+interface Props {
+  setValue: UseFormSetValue<any>;
+  fieldName: string;
+  textName: string;
+  disabled: boolean;
+}
+
+const UserFormField: React.FC<Props> = ({ setValue, fieldName, textName, disabled }: Props) => {
```
Comment (hardkov) [inline react-app/src/components/Profile/UserForm/UserFormDate.tsx:19]: Change component name to match filename.

Comment (hardkov) [inline react-app/src/components/Profile/UserForm/UserFormDate.tsx]: Change component name to match file name.

**Review by hardkov — CHANGES_REQUESTED** [review]

File: react-app/src/components/UserInsertForm.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+      <form className={classes.form} noValidate onSubmit={handleSubmit}>
+        <Grid container justify="space-between" spacing={2}>
+          <UserFormField register={register} fieldName="personalData.firstname" textName="Name" disabled={disabled} />
+          <UserFormField register={register} fieldName="personalData.lastname" textName="Surname" disabled={disabled} />
+          <UserFormField register={register} fieldName="personalData.email" textName="Email" disabled={disabled} />
+          <UserFormField register={register} fieldName="personalData.phoneNumber" textName="Phone" disabled={disabled} />
+          <UserFormField register={register} fieldName="personalData.address" textName="Address" disabled={disabled} />
+          <UserFormDate setValue={setValue} fieldName="personalData.birthdate" textName="Birthdate" disabled={disabled} />
+          <UserFormField register={register} fieldName="username" textName="Username" disabled={disabled} />
+          <UserFormField register={register} fieldName="password" textName="Password" disabled={disabled} />
```
Comment (Pinioo) [inline react-app/src/components/UserInsertForm.tsx]: Password input should be of "password" type

File: spring-app/src/main/java/com/agh/hr/persistence/service/UserService.java (hunk 1 above)
Comment (Pinioo) [reply spring-app/src/main/java/com/agh/hr/persistence/service/UserService.java]: done

File: spring-app/src/main/java/com/agh/hr/persistence/service/UserService.java (hunk 2 above)
Comment (Pinioo) [reply spring-app/src/main/java/com/agh/hr/persistence/service/UserService.java]: done

File: react-app/src/components/Profile/UserForm/UserFormDate.tsx (hunk 3 above)
Comment (Pinioo) [reply react-app/src/components/Profile/UserForm/UserFormDate.tsx]: done

**Review by ChrisMisan — APPROVED** [review]

**Review by hardkov — APPROVED** [review]
Looks good.

**Pull request merged**